build/main.mk

```diff
@@ -259,8 +259,8 @@ CXXFLAGS:= -g -fPIC
 INCLUDE_INTERNAL:=$(sort $(foreach header,$(HEADERS_FULL),$(patsubst %/,%,$(dir $(header)))))
 INCLUDE_INTERNAL+=$(BUILD_PATH)/config
 COMPILE_FLAGS:=$(addprefix -I,$(QT_HEADER_DIRS) $(INCLUDE_INTERNAL) $(GEN_SRC_PATH))
-# Enable C++11
-COMPILE_FLAGS+=-std=c++11
+# Enable C++17
+COMPILE_FLAGS+=-std=c++17
```

Comment:
I was considering that everything was going to be squashed anyway, so I didn't bother putting them in order. But this is easy to fix, since each commit changes files separately.

Reply:
I very much appreciate that you split your PR in smaller chunks. This makes it easier to review. And also later, if we find a problem, we can go back and find exactly which commit introduced the problem.
But of course the latter only works if each commit in the series compiles and works.
Please try to refactor your PR in such a form. Thanks.

Reply:
Fixed